internal/mode/static/state/graph/backend_refs.go

```diff
@@ -127,16 +144,108 @@ func createBackendRef(
 		return backendRef, &cond
 	}
 
+	backendTLSPolicy, err := findBackendTLSPolicyForService(
+		backendTLSPolicies,
+		ref,
+		sourceNamespace,
+		configMapResolver,
+	)
+	if err != nil {
+		backendRef = BackendRef{
+			Weight: weight,
+			Valid:  false,
+		}
+
+		cond := staticConds.NewRouteBackendRefUnsupportedValue(err.Error())
+		return backendRef, &cond
+	}
+
 	backendRef = BackendRef{
-		Svc:    svc,
-		Port:   port,
-		Valid:  true,
-		Weight: weight,
+		Svc:              svc,
+		BackendTLSPolicy: backendTLSPolicy,
+		Port:             port,
+		Valid:            true,
+		Weight:           weight,
 	}
 
 	return backendRef, nil
 }
 
+func findBackendTLSPolicyForService(
+	backendTLSPolicies map[types.NamespacedName]*gatewayv1alpha2.BackendTLSPolicy,
```

Comment:
I wonder if we already pass here a map between services and backend policies, which we can build separately. During that building, we can also resolve any conflicts.